pipeline_dp/dp_engine.py

```diff
@@ -338,12 +361,18 @@ def _select_private_partitions_internal(self, col,
             mechanism_type=pipeline_dp.MechanismType.GENERIC)
 
         def filter_fn(
-            budget: 'MechanismSpec', max_partitions: int,
+            budget: 'MechanismSpec', max_partitions: int, max_privacy_ids_per_row: int,
             row: Tuple[Any,
                        combiners.CompoundCombiner.AccumulatorType]) -> bool:
             """Lazily creates a partition selection strategy and uses it to determine which
             partitions to keep."""
-            privacy_id_count, _ = row[1]
+            row_count, _ = row[1]
+
+            # A conservative (lower) estimate of how many privacy IDs contributed to
+            # this partition. This estimate is only needed when privacy IDs are not available
+            # in the original dataset.
+            privacy_id_count = row_count // max_privacy_ids_per_row
```

Comment:
`//` rounds down, but we can round up, example

`max_privacy_ids_per_row = 2`, `row_count = 3`, then there is at least 2 privacy ids.

There is a trick for making round up for integers:
`(a + b -1) // b`.
Probably better to extract it to a separate method.